webrtc.html

```diff
@@ -3124,30 +3124,51 @@ <h4>
-                          defined in [[!RFC3986]] and obtain the <var>scheme
-                          name</var>. If the parsing based on the syntax
-                          defined in [[!RFC3986]] fails, [= exception/throw =]
-                          a {{SyntaxError}}. If the <var>scheme name</var> is
-                          not implemented by the browser [= exception/throw =]
-                          a {{NotSupportedError}}. If <var>scheme name</var> is
-                          <code class="scheme">turn</code> or <code class=
-                          "scheme">turns</code>, and parsing the <var>url</var>
-                          using the syntax defined in [[!RFC7065]] fails, [=
-                          exception/throw =] a {{SyntaxError}}. If <var>scheme
-                          name</var> is <code class="scheme">stun</code> or
-                          <code class="scheme">stuns</code>, and parsing the
-                          <var>url</var> using the syntax defined in
-                          [[!RFC7064]] fails, [= exception/throw =] a
-                          {{SyntaxError}}.
-                        </p>
+                        <p>Let <var>parsedURL</var> be the result of
+                        <a data-cite="!url#concept-url-parser">parsing</a>
+                        <var>url</var> with <code>null</code>.</p>
+                      </li>
+                      <li>
+                        <p>If <var>parsedURL</var> is failure, then
+                        [= exception/throw =] a {{SyntaxError}}.
+                      </li>
+                      <li>
+                        <p>If <var>parsedURL</var>
+                        <a data-cite="!url#include-credentials">includes
+                        credentials</a>, then [= exception/throw =] a
+                        {{SyntaxError}}.
+                      </li>
+                      <li>
+                        <p>Let <var>scheme</var> be <var>parsedURL</var>'s
+                        <a data-cite="!url#concept-url-scheme">scheme</a>.</p>
+                      </li>
+                      <li>
+                        <p>If <var>scheme</var> is neither <code>stun</code>,
+                        <code>stuns</code>, <code>turn</code>, nor
+                        <code>turns</code>, then [= exception/throw =] a
+                        {{SyntaxError}}.</p>
+                      </li>
+                      <li>
+                        <p>If <var>scheme</var> is not implemented by the
+                        user agent, then [= exception/throw =]
+                        a {{NotSupportedError}}.
+                      </li>
+                      <li>
+                        <p>If <var>scheme</var> is <code>stun</code> or
+                        <code>stuns</code>, and <var>parsedURL</var> violates
+                        the syntax defined in [[!RFC7064]] section 3.1,
+                        then [= exception/throw =] a {{SyntaxError}}.</p>
```

Comment:
This doesn't work as parsedURL is a data structure. How is this implemented?

Reply:
In Firefox, this is implemented [here](https://searchfox.org/mozilla-central/rev/d21e359bd26dd0a7ba216472184d6fed8f0afd48/dom/media/webrtc/jsapi/MediaTransportHandler.cpp#302-306).

Reply:
That URL isn't hierarchical though, is it? Hierarchical URLs contain slashes, as far as I know.

It sounds like you want https://url.spec.whatwg.org/#url-opaque-path to be true, although you wouldn't have to check for credentials in that case.

And then as per the Firefox code you'd have to parse parsedURL's path (which will be a string in this case) in a custom manner to obtain the host and port (split on `:`, then handle each?). Potentially this logic could be shared with the URL Standard in some way.

And then you'd handle the parsedURL's query as per Firefox logic for the query.